src/parser/metrics/Classes.ts

```diff
@@ -14,7 +17,27 @@ export class Classes implements Metric {
     private readonly statementsSuperSet: QueryStatementInterface[] = [];
 
     constructor(allNodeTypes: ExpressionMetricMapping[]) {
-        this.statementsSuperSet = getQueryStatements(allNodeTypes, this.getName());
+        for (const nodeType of allNodeTypes) {
+            if (nodeType.metrics.includes(this.getName())) {
+                if (nodeType.expression === "type_alias_declaration") {
+                    this.statementsSuperSet.push(
+                        new SimpleLanguageSpecificQueryStatement(
+                            "(type_alias_declaration value: (object_type))",
+                            nodeType.languages,
+                            nodeType.activated_for_languages,
+                        ),
+                    );
```

Comment:
Do you expect that this specific `(type_alias_declaration value: (object_type))` can also occur in other languages? Otherwise, we might just use getQueryStatements(...), remove "classes" from "type_alias_declaration" in the nodeTypesConfig.json again and add the custom query for ts and tsx afterwards. I have done it that way for the custom queries for the C and C++ classes metric currently, we might want to do that consistently.

Reply:
I thought about the same thing. I think, it is fine like this. With this approach it would be possible to reuse this logic for other languages that have a type_alias_decleration as well.

Reply:
The benefit is that we can configure in the nodeTypesConfig for which languages we use this custom query. 
The disadvantage would be that we can then only treat `type_alias_declaration` with this custom query for ts and tsx now, which is unlikely to be applicable to any other language.

Reply:
We now settled on not doing this via the nodeTypesConfig. It would be really intransparent to "add" a node here to a metric for a language but then some special handling kind of "catches" that and does something unexpected that was programmed to work for one language only. For the unlikely case that someone wants to add another language to be handled by this exact same special logic, it can be added in the source code.